pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -152,6 +154,33 @@ func (r *ReferenceAdapter) updateProjectID() error {
 	return r.kubeClient.Update(context.TODO(), r.projectReference)
 }
 
+// IsDeletionRequested checks the metadata.deletionTimestamp of ProjectReference instance, and returns if delete requested.
+// The controllers watching the ProjectReference use this as a signal to know when to execute the finalizer.
+func (r *ReferenceAdapter) IsDeletionRequested() bool {
+	return r.projectReference.DeletionTimestamp != nil
+}
+
+// AddPrjRefFinalizer parses the meta.Finalizers of ProjectReference instance and adds prjRefFinalizer if not found.
+// It returns a modification to the ProjectReference instance in the cluster for the controller to handle the reconciliation.
+func (r *ReferenceAdapter) AddPrjRefFinalizer() error {
```

Comment:
In know names tend to get lengthy, but that shouldn't be an issue with a decent IDE, so I'd prefer to not use abbreviations as they may make the code harder to understand. In this case, I even think it's obvious that we're updating a Finalizer of a ProjectReference.

Reply:
Also, the name suggests we're always adding a finalizer, but actually it _ensures it exists_.